MetaMask/metamask-extension
# Reduce calls of the `/featureFlag` API

## Explanation
Instead of periodically checking if the Swaps feature is enabled every 10 minutes, we will check it only when a user enters the Swaps feature. That will dramatically decrease calls to the `/featureFlag` API.

## Manual testing steps
Feature is enabled:
- Open the MetaMask extension
- You should see the "Swap" button every time now
- Click on the "Swap" button
- Page for showing the Swaps feature is displayed immediately
- A user selects From and To tokens and is able to successfully swap them

Feature is enabled - loaded from 10 min cache without any API call to the `/featureFlag` API:
- Finish the first test case ("Feature is enabled")
- Open the MetaMask extension
- You should see the "Swap" button every time now
- Click on the "Swap" button
- Page for showing the Swaps feature is displayed immediately, API call to the `/featureFlag` API is loaded from our 10 min cache and a user is still on the Swaps page
- A user selects From and To tokens and is able to successfully swap them

Feature is disabled:
- In ui/app/ducks/swaps/swaps.js -> fetchSwapsLiveness dispatch and return `false`
- Open the MetaMask extension
- You should see the "Swap" button every time now
- Click on the "Swap" button
- Page for showing the Swaps feature is displayed immediately with a disabled "Review Swap" button
- The "Offline for maintenance" error page is displayed almost immediately after the `/featureFlag` call is done
- A user can click on the "Close" button / Cancel or the Fox head, get back to the main page and try again. However, we cache responses for 10 minutes, so they will need to wait a bit before trying again

## Screenshots
Swap button is always visible now:
<img width="353" alt="image" src="https://user-images.githubusercontent.com/80175477/114111879-f5695b80-988f-11eb-97b4-787845223e27.png">

Swap page is accessible immediately by default:
<img width="341" alt="image" src="https://user-images.githubusercontent.com/80175477/114111999-3b262400-9890-11eb-8638-81bbce2c7394.png">

Show the "Offline for maintenance" page if the Swaps feature is disabled:
<img width="345" alt="image" src="https://user-images.githubusercontent.com/80175477/114112049-5c871000-9890-11eb-8ea3-aea4a624036c.png">
